rasa/utils/train_utils.py

```diff
@@ -557,12 +560,32 @@ def _check_loss_setting(component_config: Dict[Text, Any]) -> None:
         SOFTMAX,
         CROSS_ENTROPY,
     ]:
-        rasa.shared.utils.io.raise_warning(
+        rasa.shared.utils.io.raise_deprecation_warning(
             f"{CONSTRAIN_SIMILARITIES} is set to `False`. It is recommended "
             f"to set it to `True` when using cross-entropy loss. It will be set to "
```

Comment:
You'll need to update these functions too - 
`update_similarity_type`
`_check_similarity_loss_setting`

Also set `similarity_type` to `cosine` if `loss_type=triplet` and `similarity_type=auto`.